test/unit/test_tricrypto.py

```diff
@@ -435,3 +435,38 @@ def test_dydxfee(vyper_tricrypto):
         dx *= precisions[i]
         dy *= precisions[j]
         assert abs(dydx - dy / dx) / (dy / dx) < 1e-4
+
+
+@given(positive_balance, positive_balance, positive_balance)
+@settings(
+    suppress_health_check=[HealthCheck.function_scoped_fixture],
+    max_examples=5,
+    deadline=None,
+    phases=[Phase.reuse, Phase.generate, Phase.target],
+)
+def test_calc_token_amount(vyper_tricrypto, x0, x1, x2):
+    """
+    Test `calc_token_amount` against vyper implementation.
+
+    CurveCryptoPool and tricrypto_views.vy have slightly different
+    _fee implementations, causing a 1-2 wei difference in LP token
+    amounts calculated.
+    """
+    n_coins = 3
+    xp = [x0, x1, x2]
+    assume(0.02 < xp[0] / xp[1] < 50)
+    assume(0.02 < xp[1] / xp[2] < 50)
+    assume(0.02 < xp[0] / xp[2] < 50)
+
+    precisions = vyper_tricrypto.precisions()
+    price_scale = [vyper_tricrypto.price_scale(i) for i in range(n_coins - 1)]
+    amounts = get_real_balances(xp, precisions, price_scale)
+
+    pool = initialize_pool(vyper_tricrypto)
+
+    expected_lp_amount = vyper_tricrypto.calc_token_amount(amounts, True)
+    lp_amount = pool.calc_token_amount(amounts)
+    assert abs(lp_amount - expected_lp_amount) < 2
+
+    expected_balances = [vyper_tricrypto.balances(i) for i in range(n_coins)]
+    assert pool.balances == expected_balances
```

Comment:
I think this is to check the balances didn't mutate?  It is more explicit to just check it against the old balances (which you could name appropriately) .. this is equiv but you have to also think about the vyper contract being a view.